tests/lib/eslint.js

```diff
@@ -2112,6 +2112,113 @@ describe("eslint", function() {
         });
     });
 
+    describe("Unicode BOM:", function() {
+        describe("if a text has BOM, SourceCode", function() {
+            var sourceCode;
+
+            beforeEach(function() {
+                eslint.verify("\uFEFFconsole.log('hello');");
+                sourceCode = eslint.getSourceCode();
+            });
+
+            it("should has true at `hasBOM` property.", function() {
+                assert.equal(sourceCode.hasBOM, true);
+            });
+
+            it("should not has BOM in `text` property.", function() {
+                assert.equal(sourceCode.text, "console.log('hello');");
+            });
+        });
+
+        describe("if a text doesn't have BOM, SourceCode", function() {
+            var sourceCode;
+
+            beforeEach(function() {
+                eslint.verify("console.log('hello');");
+                sourceCode = eslint.getSourceCode();
+            });
+
+            it("should has false at `hasBOM` property.", function() {
+                assert.equal(sourceCode.hasBOM, false);
+            });
+
+            it("should not has BOM in `text` property.", function() {
+                assert.equal(sourceCode.text, "console.log('hello');");
+            });
+        });
+
+        describe("if a text has BOM, message columns of the first line", function() {
+            var messages;
+
+            beforeEach(function() {
+                messages = eslint.verify(
+                    "\uFEFFconsole.log('hello');\nconsole.log('hello');",
+                    {rules: {"no-undef": 2}});
+            });
+
+            it("should be correct", function() {
+                assert.equal(messages.length, 2);
+                assert.equal(messages[0].line, 1);
+                assert.equal(messages[0].column, 2);
+                assert.equal(messages[1].line, 2);
+                assert.equal(messages[1].column, 1);
+            });
+        });
+
+        describe("if a text doesn't have BOM, message columns of the first line", function() {
+            var messages;
+
+            beforeEach(function() {
+                messages = eslint.verify(
+                    "console.log('hello');\nconsole.log('hello');",
+                    {rules: {"no-undef": 2}});
+            });
+
+            it("should be correct", function() {
+                assert.equal(messages.length, 2);
+                assert.equal(messages[0].line, 1);
+                assert.equal(messages[0].column, 1);
+                assert.equal(messages[1].line, 2);
+                assert.equal(messages[1].column, 1);
+            });
+        });
+
+        // Skip in browsers because we cannot use "fs".
+        (typeof window === "object" ? xdescribe : describe)("when it read a UTF-8 file (has BOM), SourceCode", function() {
```

Comment:
Should I separate test file for this?

The purpose of this test is to confirm that it can handle UTF-8 BOM `[0xEF, 0xBB, 0xBF]` in a file.
In other tests, those are using UTF-16 BOM in a string literal.